### DISCUSSION TITLE
Teach ChatGPT Alda syntax, and let it guide us to compose

### DISCUSSION THREAD

**Discussion by mazzzystar** [alda-lang/alda#472] [Ideas]
Hey there! So, I had a weird idea, "Teach, Show, Then Consult", which is to teach ChatGPT Alda language syntax, show it some superb code, finally consulting with it to co-create a beautiful melody together! You can find in [article](https://mazzzystar.github.io/2023/03/03/teach-show-consult-gpt/) or [code](https://github.com/mazzzystar/teach-show-consult) for more detail explaination.

Hope you guys find it's fun : )

**Comment by daveyarwood** [comment]
Wow, I love this! :heart:  ChatGPT + Alda seems like a really promising way to produce generative music. I'm already amazed at the result you posted in your article. It actually compiles, and it sounds pretty good/interesting. I only wish there were a larger corpus of Alda scores on the internet that we could feed it, with context about each score to guide ChatGPT about how to produce different kinds of music.

**Reply by doodead** [reply]
Well, if that helps somehow, I can take a bunch of compositions of mine and transcribe them to beautiful Alda (I mean, the Alda will be beautiful, not my compositions 😂 ). I can probably try to get a feel to what would be "standard Alda" by looking at the examples we have in the repo itself (I did look at them before to learn how to Alda but I tend to take much more liberties because of my specific use cases, but I would aim for public disclosure levels of quality and not personal notes levels of quality if the idea is to contribute in a meaningful way to an AI model's knowledge).

I don't know if the current "corpus" of Alda scores is along the lines of 10s, 100s, 1000s or more (because if there's some 100s or more maybe half a dozen from me wouldn't be all that much), but yeah, I follow Alda for some years already and though I myself hadn't have the time to do nearly as much with it as I would like, I still feel like there's so much potential to integrate it with other tools, even if just simple personal scripts, to make the process of writing music easier. Would be cool to try to incorporate it into an advanced language model.
